iati/validator.py

```diff
@@ -324,20 +485,27 @@ def is_valid(dataset, schema):
 def is_xml(maybe_xml):
     """Determine whether a given parameter is XML.
 
-
     Args:
         maybe_xml (str): An string that may or may not contain valid XML.
 
     Returns:
         bool: A boolean indicating whether the given Dataset is valid XML.
 
     """
-    if isinstance(maybe_xml, iati.core.data.Dataset):
-        maybe_xml = maybe_xml.xml_str
+    error_log = _check_is_xml(maybe_xml)
 
-    try:
-        _ = etree.fromstring(maybe_xml.strip())
-        return True
-    except (etree.XMLSyntaxError, AttributeError, TypeError, ValueError):
-        return False
+    return not error_log.contains_errors()
+
+
+def validate_is_xml(dataset):
+    """Check whether a Dataset contains valid XML.
+
+    Args:
+        dataset (iati.core.Dataset): The Dataset to check validity of.
+
+    Returns:
+        iati.validator.ValidationErrorLog: A log of the errors that occurred.
+
+    """
+    return _check_is_xml(dataset)
```

Comment:
`_check_is_xml` takes a `str`, but this seems to be passing it a `iati.core.Dataset`.